debezium-core/src/main/java/io/debezium/transforms/TimezoneConverter.java

```diff
@@ -307,12 +308,38 @@ private void handleStructs(Struct value, Type type, String matchName, Set<String
 
         Struct before = getStruct(value, Envelope.FieldName.BEFORE);
         Struct after = getStruct(value, Envelope.FieldName.AFTER);
+        Struct source = getStruct(value, Envelope.FieldName.SOURCE);
+
+        Set<String> beforeFields = new HashSet<>();
+        Set<String> afterFields = new HashSet<>();
+        Set<String> sourceFields = new HashSet<>();
+
+        if (!fields.isEmpty() && !fields.contains(null)) {
+            for (String field : fields) {
+                if (field.startsWith(FIELD_SOURCE_PREFIX)) {
```

Comment:
I think you should match for `source.` includeing the dot, not only `source` as it would conflict with allowed field names.
Same for before/after below.